internal/orchestration/controller.go

```diff
@@ -342,14 +343,31 @@ func (e *Experiment) stopReplicas() error {
 		for id, hash := range res.GetHashes() {
 			hashes[id] = hash
 		}
+		for id, cnt := range res.GetCounts() {
```

Comment:
What are Lines 343-348 doing? Merging the final hashes and command counts from all replicas? Won't this overwrite the hash/count value for a given replica's `id`? 

My reading of the code here is that `worker.StopReplica` is called on all replicas, each of which will return the map of their hashes and counts for each of their counterparts. Presumably, these will have overlapping `id` fields and so would overwrite the same `id` entry. What am I missing here?